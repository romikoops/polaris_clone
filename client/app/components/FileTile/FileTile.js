import React from 'react';
import fetch from 'isomorphic-fetch';
import PropTypes from 'prop-types';
import { Promise } from 'es6-promise-promise';
import { BASE_URL } from '../../constants';
import { authHeader } from '../../helpers';
import styles from './FileTile.scss';
// import { RoundButton } from '../RoundButton/RoundButton';
import { moment, documentTypes } from '../../constants';
import {Link} from 'react-router-dom';
const docTypes = documentTypes;
class FileTile extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            file: null
        };
        this.onFormSubmit = this.onFormSubmit.bind(this);
        this.onChange = this.onChange.bind(this);
        this.fileUpload = this.fileUpload.bind(this);
        this.deleteFile = this.deleteFile.bind(this);
    }
    handleResponse(response) {
        if (!response.ok) {
            return Promise.reject(response.statusText);
        }

        return response.json();
    }
    onFormSubmit(e) {
        e.preventDefault(); // Stop form submit
        this.fileUpload(this.state.file).then((response)=>{
            console.log(response.data);
        });
    }
    onChange(e) {
        // this.setState({file: e.target.files[0]});
        this.fileUpload(e.target.files[0]);
    }
    // Delete the file
    deleteFile() {
        const {doc, deleteFn} = this.props;
        deleteFn(doc.id);
    }
    fileUpload(file) {
        const {type, dispatchFn, doc} = this.props;
        const url = '/shipments/' + doc.shipment_id + '/upload/' + doc.doc_type;
        if (!file) {
            return '';
        }
        if (dispatchFn) {
            return dispatchFn(file);
        }
        const formData = new FormData();
        formData.append('file', file);
        formData.append('type', type);
        const requestOptions = {
            method: 'POST',
            headers: { ...authHeader()},
            body: formData
        };
        const uploadUrl = BASE_URL + url;
        return fetch(uploadUrl, requestOptions).then(this.handleResponse);
    }

    render() {
        const clickUploaderInput = () => {
            this.uploaderInput.click();
        };
        const {theme, type, doc} = this.props;
        const textStyle = {
            background: theme && theme.colors ? '-webkit-linear-gradient(left, ' + theme.colors.primary + ',' + theme.colors.secondary + ')' : 'black'
        };
        const link = doc.signed_url ?
            (<Link to={doc.signed_url} className="flex-none layout-row layout-align-center-center" target="_blank">
                <i className="clip fa fa-eye" style={textStyle}></i>
            </Link>) :
            '';
        return (
            <div className={`flex-none layout-row layout-wrap layout-align-center-start ${styles.tile}`}>
                <div className="flex-100 layout-row layout-wrap layout-align-center-center">
                    <div className="flex-100 layout-row layout-wrap layout-align-center-start">
                        <div className={`flex-100 layout-row layout-wrap layout-align-center-start ${styles.file_header}`}>
                            <p className="flex-100">Title</p>
                        </div>
                        <div className={`flex-100 layout-row layout-wrap layout-align-center-start ${styles.file_text}`}>
                            <p className="flex-100">{doc.text}</p>
                        </div>
                    </div>
                    <div className="flex-100 layout-row layout-wrap layout-align-center-start">
                        <div className={`flex-100 layout-row layout-wrap layout-align-center-start ${styles.file_header}`}>
                            <p className="flex-100">Type</p>
                        </div>
                        <div className={`flex-100 layout-row layout-wrap layout-align-center-start ${styles.file_text}`}>
                            <p className="flex-100">{docTypes[doc.doc_type]}</p>
                        </div>
                    </div>
                    <div className="flex-100 layout-row layout-wrap layout-align-center-start">
                        <div className={`flex-100 layout-row layout-wrap layout-align-center-start ${styles.file_header}`}>
                            <p className="flex-100">Uploaded</p>
                        </div>
                        <div className={`flex-100 layout-row layout-wrap layout-align-center-start ${styles.file_text}`}>
                            <p className="flex-100">{moment(doc.created_at).format('lll')}</p>
                        </div>
                    </div>
                </div>
                <div className="flex-100 layout-row layout-align-center-end">
                    <div className={`${styles.upload_btn_wrapper} flex-33 layout-row layout-align-center-center`}>
                        <form className="flex-none layout-row layout-align-center-center" onSubmit={this.onFormSubmit}>
                            <div className="flex-none" onClick={clickUploaderInput}>
                                <i className="fa fa-pencil clip" style={textStyle}></i>
                            </div>
                            <input type="file" onChange={this.onChange} name={type} ref={input => { this.uploaderInput = input; }}/>
                        </form>
                    </div>
                    <div className={`${styles.upload_btn_wrapper} flex-33 layout-row layout-align-center-center`}>
                        <div className="flex-none layout-row layout-align-center-center" onClick={this.deleteFile} >
                            <i className="clip fa fa-trash" style={textStyle}></i>
                        </div>
                    </div>
                    <div className={`${styles.upload_btn_wrapper} flex-33 layout-row layout-align-center-center`}>
                        {link}
                    </div>
                </div>

            </div>
        );
    }
}

FileTile.propTypes = {
    url: PropTypes.string,
    text: PropTypes.string,
    type: PropTypes.string,
    theme: PropTypes.object,
    dispatchFn: PropTypes.func
};


export default FileTile;
