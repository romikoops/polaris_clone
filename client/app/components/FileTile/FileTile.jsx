import React from 'react'
import { withNamespaces } from 'react-i18next'
import fetch from 'isomorphic-fetch'
import { Link } from 'react-router-dom'
import Truncate from 'react-truncate'
import { Promise } from 'es6-promise-promise'
import PropTypes from '../../prop-types'
import { authHeader } from '../../helpers'
import styles from './FileTile.scss'
import { RoundButton } from '../RoundButton/RoundButton'
import { moment, documentTypes } from '../../constants'
import getApiHost from '../../constants/api.constants'

const docTypes = documentTypes
class FileTile extends React.Component {
  static handleResponse (response) {
    if (!response.ok) {
      return Promise.reject(response.statusText)
    }

    return response.json()
  }
  constructor (props) {
    super(props)
    this.state = {
      file: null,
      denial: {},
      showDenialDetails: false
    }
    this.onFormSubmit = this.onFormSubmit.bind(this)
    this.onChange = this.onChange.bind(this)
    this.fileUpload = this.fileUpload.bind(this)
    this.deleteFile = this.deleteFile.bind(this)
    this.toggleShowDenial = this.toggleShowDenial.bind(this)
    this.handleDeny = this.handleDeny.bind(this)
    this.handleApprove = this.handleApprove.bind(this)
    this.handleDenialForm = this.handleDenialForm.bind(this)
  }
  onFormSubmit (e) {
    e.preventDefault()
    this.fileUpload(this.state.file)
  }
  onChange (e) {
    this.fileUpload(e.target.files[0])
  }
  deleteFile () {
    const { doc, deleteFn } = this.props
    deleteFn(doc.id)
  }
  handleDenialForm (ev) {
    const { value } = ev.target
    this.setState({ denial: { text: value } })
  }
  fileUpload (file) {
    const { type, dispatchFn, doc } = this.props
    const url = `/shipments/${doc.shipment_id}/upload/${doc.doc_type}`
    if (!file) {
      return ''
    }
    if (dispatchFn) {
      return dispatchFn(file)
    }
    const formData = new window.FormData()
    formData.append('file', file)
    formData.append('type', type)
    const requestOptions = {
      method: 'POST',
      headers: { ...authHeader() },
      body: formData
    }
    if (this.uploaderInput.files.length) {
      this.uploaderInput.value = ''
    }
    const uploadUrl = getApiHost() + url

    return fetch(uploadUrl, requestOptions).then(FileTile.handleResponse)
  }
  handleDeny () {
    const { doc, adminDispatch } = this.props
    const { denial } = this.state
    denial.type = 'reject'
    adminDispatch.documentAction(doc.id, denial)
    this.toggleShowDenial()
  }
  handleApprove () {
    const { doc, adminDispatch } = this.props
    const { denial } = this.state
    denial.type = 'approve'
    adminDispatch.documentAction(doc.id, denial)
  }
  toggleShowDenial () {
    this.setState({ showDenialDetails: !this.state.showDenialDetails })
  }

  render () {
    const clickUploaderInput = () => {
      this.uploaderInput.click()
    }
    const {
      theme, type, doc, isAdmin, t
    } = this.props
    const { showDenialDetails, denial } = this.state
    const textStyle = {
      background:
        theme && theme.colors
          ? `-webkit-linear-gradient(left, ${theme.colors.primary},${theme.colors.secondary})`
          : 'black'
    }
    let statusStyle
    if (doc.approved === 'approved') {
      statusStyle = styles.approved
    } else if (doc.approved === 'rejected') {
      statusStyle = styles.rejected
    } else if (doc.approved === null) {
      statusStyle = styles.pending
    }
    const link = doc.signed_url ? (
      <Link
        to={doc.signed_url}
        className="flex-none layout-row layout-align-center-center"
        target="_blank"
      >
        <i className="clip fa fa-eye" style={textStyle} />
      </Link>
    ) : (
      ''
    )
    const denyDetails = (
      <div className={`flex-none layout-row layout-align-center-center  ${styles.backdrop}`}>
        <div className={`flex-none ${styles.fade}`} onClick={this.toggleShowDenial} />
        <div
          className={`flex-none layout-row layout-wrap layout-align-center-start  ${
            styles.content
          }`}
        >
          <div className="flex-100 layout-row layout-align-start-center">
            <h3 className="flex-none clip" style={textStyle}>
              {t('doc:reject')}
            </h3>
          </div>
          <div className={`flex-100 layout-row layout-align-start-center ${styles.input_box}`}>
            <textarea
              rows="4"
              className="flex-100"
              value={denial.text}
              onChange={this.handleDenialForm}
            />
          </div>
          <div className="flex-100 layout-row layout-align-end-end">
            <div className="flex-none layout-row" style={{ margin: '15px' }}>
              <RoundButton
                theme={theme}
                size="small"
                text={t('common:deny')}
                iconClass="fa-times"
                handleNext={this.handleDeny}
              />
            </div>
          </div>
        </div>
      </div>
    )
    const userRow = (
      <div className="flex-100 layout-row layout-align-center-end">
        <div
          className={`${styles.upload_btn_wrapper} flex-33 layout-row layout-align-center-center`}
        >
          <form
            className="flex-none layout-row layout-align-center-center"
            onSubmit={this.onFormSubmit}
          >
            <div className="flex-none" onClick={clickUploaderInput}>
              <i className="fa fa-pencil clip" style={textStyle} />
            </div>
            <input
              type="file"
              onChange={this.onChange}
              name={type}
              ref={(input) => {
                this.uploaderInput = input
              }}
            />
          </form>
        </div>
        <div
          className={`${styles.upload_btn_wrapper} flex-33 layout-row layout-align-center-center`}
        >
          <div
            className="flex-none layout-row layout-align-center-center"
            onClick={this.deleteFile}
          >
            <i className="clip fa fa-trash" style={textStyle} />
          </div>
        </div>
        <div
          className={`${styles.upload_btn_wrapper} flex-33 layout-row layout-align-center-center`}
        >
          {link}
        </div>
      </div>
    )
    const adminRow = (
      <div className="flex-100 layout-row layout-align-center-end">
        <div
          className={`${styles.upload_btn_wrapper} flex-33 layout-row layout-align-center-center`}
        >
          <div
            className="flex-none layout-row layout-align-center-center"
            onClick={this.handleApprove}
          >
            <i className="clip fa fa-check" style={textStyle} />
          </div>
        </div>
        <div
          className={`${styles.upload_btn_wrapper} flex-33 layout-row layout-align-center-center`}
        >
          <div
            className="flex-none layout-row layout-align-center-center"
            onClick={this.toggleShowDenial}
          >
            <i className=" fa fa-times" style={{ color: 'red' }} />
          </div>
        </div>
        <div
          className={`${styles.upload_btn_wrapper} flex-33 layout-row layout-align-center-center`}
        >
          {link}
        </div>
      </div>
    )
    const bottomRow = isAdmin ? adminRow : userRow

    return (
      <div className={`flex-none layout-row layout-wrap layout-align-center-start ${styles.tile} `}>
        {showDenialDetails ? denyDetails : ''}
        <div className="flex-100 layout-row layout-wrap layout-align-center-center">
          <div className="flex-100 layout-row layout-wrap layout-align-center-start">
            <div
              className={`flex-100 layout-row layout-wrap layout-align-center-start ${
                styles.file_header
              }`}
            >
              <p className="flex-100">{t('common:title')}</p>
            </div>
            <div
              className={`flex-100 layout-row layout-wrap layout-align-center-start ${
                styles.file_text
              }`}
            >
              <p className="flex-100">
                <Truncate lines={1}>{doc.text} </Truncate>
              </p>
            </div>
          </div>
          <div className="flex-100 layout-row layout-wrap layout-align-center-start">
            <div
              className={`flex-100 layout-row layout-wrap layout-align-center-start ${
                styles.file_header
              }`}
            >
              <p className="flex-100">{t('doc:type')}</p>
            </div>
            <div
              className={`flex-100 layout-row layout-wrap layout-align-center-start ${
                styles.file_text
              }`}
            >
              <p className="flex-100">{docTypes[doc.doc_type]}</p>
            </div>
          </div>
          <div className="flex-100 layout-row layout-wrap layout-align-center-start">
            <div
              className={`flex-100 layout-row layout-wrap layout-align-center-start ${
                styles.file_header
              }`}
            >
              <p className="flex-100">{t('doc:uploaded')}</p>
            </div>
            <div
              className={`flex-100 layout-row layout-wrap layout-align-center-start ${
                styles.file_text
              }`}
            >
              <p className="flex-100">{moment(doc.created_at).format('lll')}</p>
            </div>
          </div>
          <div className="flex-100 layout-row layout-wrap layout-align-center-start">
            <div
              className={`flex-100 layout-row layout-wrap layout-align-center-start ${
                styles.file_header
              }`}
            >
              <p className="flex-100">{t('common:status')}</p>
            </div>
            <div
              className={`flex-100 layout-row layout-wrap layout-align-center-start ${
                styles.file_text
              } ${statusStyle}`}
            >
              <p className="flex-100">{doc.approved ? doc.approved : t('common:pending')}</p>
            </div>
          </div>
        </div>
        {bottomRow}
      </div>
    )
  }
}

FileTile.propTypes = {
  type: PropTypes.string.isRequired,
  t: PropTypes.func.isRequired,
  isAdmin: PropTypes.bool,
  theme: PropTypes.theme,
  dispatchFn: PropTypes.func.isRequired,
  adminDispatch: PropTypes.shape({
    documentAction: PropTypes.func
  }).isRequired,
  doc: PropTypes.shape({
    id: PropTypes.number
  }).isRequired,
  deleteFn: PropTypes.func.isRequired
}

FileTile.defaultProps = {
  theme: null,
  isAdmin: false
}

export default withNamespaces(['common', 'doc'])(FileTile)
