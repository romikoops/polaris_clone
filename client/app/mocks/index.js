export * from './address'
export * from './cards'
export * from './cargoItems'
export * from './client'
export * from './consignee'
export * from './contacts'
export * from './containers'
export * from './currencies'
export * from './dimensions'
export * from './documents'
export * from './feeHash'
export * from './functions'
export * from './gMaps'
export * from './group'
export * from './hub'
export * from './match'
export * from './notifyees'
export * from './offer'
export * from './results'
export * from './routes'
export * from './schedules'
export * from './scope'
export * from './shipmentData'
export * from './shipments'
export * from './shipper'
export * from './tenant'
export * from './theme'
export * from './user'
