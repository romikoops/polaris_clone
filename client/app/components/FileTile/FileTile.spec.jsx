import * as React from 'react'
import { shallow } from 'enzyme'
import { theme, identity } from '../../mocks'

jest.mock('isomorphic-fetch', () =>
  () => Promise.resolve({ data: [] }))
jest.mock('react-router', () => ({
  // eslint-disable-next-line react/prop-types
  Link: ({ props }) => <a {...props}>link</a>
}))
jest.mock('../../helpers', () => ({
  authHeader: x => x
}))
jest.mock('../../constants', () => {
  const format = () => 19
  const add = () => ({ format })

  const moment = () => ({
    format,
    add
  })

  const documentTypes = {
    packing_sheet: 'Packing List',
    commercial_invoice: 'Commercial Invoice',
    customs_declaration: 'Customs Declaration',
    customs_value_declaration: 'Customs Value Declaration',
    eori: 'EORI',
    certificate_of_origin: 'Certificate Of Origin',
    dangerous_goods: 'Dangerous Goods',
    bill_of_lading: 'Bill of Lading',
    invoice: 'Invoice',
    miscellaneous: 'Miscellaneous'
  }
  const getTenantApiUrl = () => 'BASE_URL'

  return { moment, documentTypes, getTenantApiUrl }
})
// eslint-disable-next-line
import FileTile from './FileTile'

const propsBase = {
  theme,
  type: 'FOO_TYPE',
  isAdmin: false,
  dispatchFn: identity,
  adminDispatch: {
    documentAction: identity
  },
  doc: {
    id: 9,
    approved: 'approved'
  },
  deleteFn: identity
}

test('shallow render', () => {
  expect(shallow(<FileTile {...propsBase} />)).toMatchSnapshot()
})

test('doc.approved === rejected', () => {
  const props = {
    ...propsBase,
    doc: {
      ...propsBase.doc,
      approved: 'rejected'
    }
  }
  expect(shallow(<FileTile {...props} />)).toMatchSnapshot()
})

test('doc.approved === null', () => {
  const props = {
    ...propsBase,
    doc: {
      ...propsBase.doc,
      approved: null
    }
  }
  expect(shallow(<FileTile {...props} />)).toMatchSnapshot()
})

test('doc.signed_url is truthy', () => {
  const props = {
    ...propsBase,
    doc: {
      ...propsBase.doc,
      signed_url: 'FOO_SIGNED_URL'
    }
  }
  expect(shallow(<FileTile {...props} />)).toMatchSnapshot()
})

test('isAdmin is true', () => {
  const props = {
    ...propsBase,
    isAdmin: true
  }
  expect(shallow(<FileTile {...props} />)).toMatchSnapshot()
})

test('theme is falsy', () => {
  const props = {
    ...propsBase,
    theme: null
  }
  expect(shallow(<FileTile {...props} />)).toMatchSnapshot()
})

test('state.showDenialDetails is true', () => {
  const wrapper = shallow(<FileTile {...propsBase} />)
  wrapper.setState({ showDenialDetails: true })

  expect(wrapper).toMatchSnapshot()
})
