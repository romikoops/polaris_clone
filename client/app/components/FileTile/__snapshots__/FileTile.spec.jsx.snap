// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`doc.approved === "rejected" 1`] = `
<div
  className="flex-none layout-row layout-wrap layout-align-center-start tile "
>
  <div
    className="flex-100 layout-row layout-wrap layout-align-center-center"
  >
    <div
      className="flex-100 layout-row layout-wrap layout-align-center-start"
    >
      <div
        className="flex-100 layout-row layout-wrap layout-align-center-start file_header"
      >
        <p
          className="flex-100"
        >
          Title
        </p>
      </div>
      <div
        className="flex-100 layout-row layout-wrap layout-align-center-start file_text"
      >
        <p
          className="flex-100"
        >
          <Component
            lines={1}
          >
             
          </Component>
        </p>
      </div>
    </div>
    <div
      className="flex-100 layout-row layout-wrap layout-align-center-start"
    >
      <div
        className="flex-100 layout-row layout-wrap layout-align-center-start file_header"
      >
        <p
          className="flex-100"
        >
          Type
        </p>
      </div>
      <div
        className="flex-100 layout-row layout-wrap layout-align-center-start file_text"
      >
        <p
          className="flex-100"
        />
      </div>
    </div>
    <div
      className="flex-100 layout-row layout-wrap layout-align-center-start"
    >
      <div
        className="flex-100 layout-row layout-wrap layout-align-center-start file_header"
      >
        <p
          className="flex-100"
        >
          Uploaded
        </p>
      </div>
      <div
        className="flex-100 layout-row layout-wrap layout-align-center-start file_text"
      >
        <p
          className="flex-100"
        >
          19
        </p>
      </div>
    </div>
    <div
      className="flex-100 layout-row layout-wrap layout-align-center-start"
    >
      <div
        className="flex-100 layout-row layout-wrap layout-align-center-start file_header"
      >
        <p
          className="flex-100"
        >
          Status
        </p>
      </div>
      <div
        className="flex-100 layout-row layout-wrap layout-align-center-start file_text rejected"
      >
        <p
          className="flex-100"
        >
          rejected
        </p>
      </div>
    </div>
  </div>
  <div
    className="flex-100 layout-row layout-align-center-end"
  >
    <div
      className="upload_btn_wrapper flex-33 layout-row layout-align-center-center"
    >
      <form
        className="flex-none layout-row layout-align-center-center"
        onSubmit={[Function]}
      >
        <div
          className="flex-none"
          onClick={[Function]}
        >
          <i
            className="fa fa-pencil clip"
            style={
              Object {
                "background": "-webkit-linear-gradient(left, #333,#fafafa)",
              }
            }
          />
        </div>
        <input
          name="FOO_TYPE"
          onChange={[Function]}
          type="file"
        />
      </form>
    </div>
    <div
      className="upload_btn_wrapper flex-33 layout-row layout-align-center-center"
    >
      <div
        className="flex-none layout-row layout-align-center-center"
        onClick={[Function]}
      >
        <i
          className="clip fa fa-trash"
          style={
            Object {
              "background": "-webkit-linear-gradient(left, #333,#fafafa)",
            }
          }
        />
      </div>
    </div>
    <div
      className="upload_btn_wrapper flex-33 layout-row layout-align-center-center"
    />
  </div>
</div>
`;

exports[`doc.approved === null 1`] = `
<div
  className="flex-none layout-row layout-wrap layout-align-center-start tile "
>
  <div
    className="flex-100 layout-row layout-wrap layout-align-center-center"
  >
    <div
      className="flex-100 layout-row layout-wrap layout-align-center-start"
    >
      <div
        className="flex-100 layout-row layout-wrap layout-align-center-start file_header"
      >
        <p
          className="flex-100"
        >
          Title
        </p>
      </div>
      <div
        className="flex-100 layout-row layout-wrap layout-align-center-start file_text"
      >
        <p
          className="flex-100"
        >
          <Component
            lines={1}
          >
             
          </Component>
        </p>
      </div>
    </div>
    <div
      className="flex-100 layout-row layout-wrap layout-align-center-start"
    >
      <div
        className="flex-100 layout-row layout-wrap layout-align-center-start file_header"
      >
        <p
          className="flex-100"
        >
          Type
        </p>
      </div>
      <div
        className="flex-100 layout-row layout-wrap layout-align-center-start file_text"
      >
        <p
          className="flex-100"
        />
      </div>
    </div>
    <div
      className="flex-100 layout-row layout-wrap layout-align-center-start"
    >
      <div
        className="flex-100 layout-row layout-wrap layout-align-center-start file_header"
      >
        <p
          className="flex-100"
        >
          Uploaded
        </p>
      </div>
      <div
        className="flex-100 layout-row layout-wrap layout-align-center-start file_text"
      >
        <p
          className="flex-100"
        >
          19
        </p>
      </div>
    </div>
    <div
      className="flex-100 layout-row layout-wrap layout-align-center-start"
    >
      <div
        className="flex-100 layout-row layout-wrap layout-align-center-start file_header"
      >
        <p
          className="flex-100"
        >
          Status
        </p>
      </div>
      <div
        className="flex-100 layout-row layout-wrap layout-align-center-start file_text pending"
      >
        <p
          className="flex-100"
        >
          Pending
        </p>
      </div>
    </div>
  </div>
  <div
    className="flex-100 layout-row layout-align-center-end"
  >
    <div
      className="upload_btn_wrapper flex-33 layout-row layout-align-center-center"
    >
      <form
        className="flex-none layout-row layout-align-center-center"
        onSubmit={[Function]}
      >
        <div
          className="flex-none"
          onClick={[Function]}
        >
          <i
            className="fa fa-pencil clip"
            style={
              Object {
                "background": "-webkit-linear-gradient(left, #333,#fafafa)",
              }
            }
          />
        </div>
        <input
          name="FOO_TYPE"
          onChange={[Function]}
          type="file"
        />
      </form>
    </div>
    <div
      className="upload_btn_wrapper flex-33 layout-row layout-align-center-center"
    >
      <div
        className="flex-none layout-row layout-align-center-center"
        onClick={[Function]}
      >
        <i
          className="clip fa fa-trash"
          style={
            Object {
              "background": "-webkit-linear-gradient(left, #333,#fafafa)",
            }
          }
        />
      </div>
    </div>
    <div
      className="upload_btn_wrapper flex-33 layout-row layout-align-center-center"
    />
  </div>
</div>
`;

exports[`props.isAdmin is true 1`] = `
<div
  className="flex-none layout-row layout-wrap layout-align-center-start tile "
>
  <div
    className="flex-100 layout-row layout-wrap layout-align-center-center"
  >
    <div
      className="flex-100 layout-row layout-wrap layout-align-center-start"
    >
      <div
        className="flex-100 layout-row layout-wrap layout-align-center-start file_header"
      >
        <p
          className="flex-100"
        >
          Title
        </p>
      </div>
      <div
        className="flex-100 layout-row layout-wrap layout-align-center-start file_text"
      >
        <p
          className="flex-100"
        >
          <Component
            lines={1}
          >
             
          </Component>
        </p>
      </div>
    </div>
    <div
      className="flex-100 layout-row layout-wrap layout-align-center-start"
    >
      <div
        className="flex-100 layout-row layout-wrap layout-align-center-start file_header"
      >
        <p
          className="flex-100"
        >
          Type
        </p>
      </div>
      <div
        className="flex-100 layout-row layout-wrap layout-align-center-start file_text"
      >
        <p
          className="flex-100"
        />
      </div>
    </div>
    <div
      className="flex-100 layout-row layout-wrap layout-align-center-start"
    >
      <div
        className="flex-100 layout-row layout-wrap layout-align-center-start file_header"
      >
        <p
          className="flex-100"
        >
          Uploaded
        </p>
      </div>
      <div
        className="flex-100 layout-row layout-wrap layout-align-center-start file_text"
      >
        <p
          className="flex-100"
        >
          19
        </p>
      </div>
    </div>
    <div
      className="flex-100 layout-row layout-wrap layout-align-center-start"
    >
      <div
        className="flex-100 layout-row layout-wrap layout-align-center-start file_header"
      >
        <p
          className="flex-100"
        >
          Status
        </p>
      </div>
      <div
        className="flex-100 layout-row layout-wrap layout-align-center-start file_text approved"
      >
        <p
          className="flex-100"
        >
          approved
        </p>
      </div>
    </div>
  </div>
  <div
    className="flex-100 layout-row layout-align-center-end"
  >
    <div
      className="upload_btn_wrapper flex-33 layout-row layout-align-center-center"
    >
      <div
        className="flex-none layout-row layout-align-center-center"
        onClick={[Function]}
      >
        <i
          className="clip fa fa-check"
          style={
            Object {
              "background": "-webkit-linear-gradient(left, #333,#fafafa)",
            }
          }
        />
      </div>
    </div>
    <div
      className="upload_btn_wrapper flex-33 layout-row layout-align-center-center"
    >
      <div
        className="flex-none layout-row layout-align-center-center"
        onClick={[Function]}
      >
        <i
          className=" fa fa-times"
          style={
            Object {
              "color": "red",
            }
          }
        />
      </div>
    </div>
    <div
      className="upload_btn_wrapper flex-33 layout-row layout-align-center-center"
    />
  </div>
</div>
`;

exports[`shallow render 1`] = `
<div
  className="flex-none layout-row layout-wrap layout-align-center-start tile "
>
  <div
    className="flex-100 layout-row layout-wrap layout-align-center-center"
  >
    <div
      className="flex-100 layout-row layout-wrap layout-align-center-start"
    >
      <div
        className="flex-100 layout-row layout-wrap layout-align-center-start file_header"
      >
        <p
          className="flex-100"
        >
          Title
        </p>
      </div>
      <div
        className="flex-100 layout-row layout-wrap layout-align-center-start file_text"
      >
        <p
          className="flex-100"
        >
          <Component
            lines={1}
          >
             
          </Component>
        </p>
      </div>
    </div>
    <div
      className="flex-100 layout-row layout-wrap layout-align-center-start"
    >
      <div
        className="flex-100 layout-row layout-wrap layout-align-center-start file_header"
      >
        <p
          className="flex-100"
        >
          Type
        </p>
      </div>
      <div
        className="flex-100 layout-row layout-wrap layout-align-center-start file_text"
      >
        <p
          className="flex-100"
        />
      </div>
    </div>
    <div
      className="flex-100 layout-row layout-wrap layout-align-center-start"
    >
      <div
        className="flex-100 layout-row layout-wrap layout-align-center-start file_header"
      >
        <p
          className="flex-100"
        >
          Uploaded
        </p>
      </div>
      <div
        className="flex-100 layout-row layout-wrap layout-align-center-start file_text"
      >
        <p
          className="flex-100"
        >
          19
        </p>
      </div>
    </div>
    <div
      className="flex-100 layout-row layout-wrap layout-align-center-start"
    >
      <div
        className="flex-100 layout-row layout-wrap layout-align-center-start file_header"
      >
        <p
          className="flex-100"
        >
          Status
        </p>
      </div>
      <div
        className="flex-100 layout-row layout-wrap layout-align-center-start file_text approved"
      >
        <p
          className="flex-100"
        >
          approved
        </p>
      </div>
    </div>
  </div>
  <div
    className="flex-100 layout-row layout-align-center-end"
  >
    <div
      className="upload_btn_wrapper flex-33 layout-row layout-align-center-center"
    >
      <form
        className="flex-none layout-row layout-align-center-center"
        onSubmit={[Function]}
      >
        <div
          className="flex-none"
          onClick={[Function]}
        >
          <i
            className="fa fa-pencil clip"
            style={
              Object {
                "background": "-webkit-linear-gradient(left, #333,#fafafa)",
              }
            }
          />
        </div>
        <input
          name="FOO_TYPE"
          onChange={[Function]}
          type="file"
        />
      </form>
    </div>
    <div
      className="upload_btn_wrapper flex-33 layout-row layout-align-center-center"
    >
      <div
        className="flex-none layout-row layout-align-center-center"
        onClick={[Function]}
      >
        <i
          className="clip fa fa-trash"
          style={
            Object {
              "background": "-webkit-linear-gradient(left, #333,#fafafa)",
            }
          }
        />
      </div>
    </div>
    <div
      className="upload_btn_wrapper flex-33 layout-row layout-align-center-center"
    />
  </div>
</div>
`;

exports[`state.showDenialDetails is true 1`] = `
<div
  className="flex-none layout-row layout-wrap layout-align-center-start tile "
>
  <div
    className="flex-none layout-row layout-align-center-center  backdrop"
  >
    <div
      className="flex-none fade"
      onClick={[Function]}
    />
    <div
      className="flex-none layout-row layout-wrap layout-align-center-start  content"
    >
      <div
        className="flex-100 layout-row layout-align-start-center"
      >
        <h3
          className="flex-none clip"
          style={
            Object {
              "background": "-webkit-linear-gradient(left, #333,#fafafa)",
            }
          }
        >
          Reject document
        </h3>
      </div>
      <div
        className="flex-100 layout-row layout-align-start-center input_box"
      >
        <textarea
          className="flex-100"
          onChange={[Function]}
          rows="4"
        />
      </div>
      <div
        className="flex-100 layout-row layout-align-end-end"
      >
        <div
          className="flex-none layout-row"
          style={
            Object {
              "margin": "15px",
            }
          }
        >
          <RoundButton
            handleNext={[Function]}
            iconClass="fa-times"
            size="small"
            text="Deny"
            theme={
              Object {
                "colors": Object {
                  "primary": "#333",
                  "secondary": "#fafafa",
                },
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="flex-100 layout-row layout-wrap layout-align-center-center"
  >
    <div
      className="flex-100 layout-row layout-wrap layout-align-center-start"
    >
      <div
        className="flex-100 layout-row layout-wrap layout-align-center-start file_header"
      >
        <p
          className="flex-100"
        >
          Title
        </p>
      </div>
      <div
        className="flex-100 layout-row layout-wrap layout-align-center-start file_text"
      >
        <p
          className="flex-100"
        >
          <Component
            lines={1}
          >
             
          </Component>
        </p>
      </div>
    </div>
    <div
      className="flex-100 layout-row layout-wrap layout-align-center-start"
    >
      <div
        className="flex-100 layout-row layout-wrap layout-align-center-start file_header"
      >
        <p
          className="flex-100"
        >
          Type
        </p>
      </div>
      <div
        className="flex-100 layout-row layout-wrap layout-align-center-start file_text"
      >
        <p
          className="flex-100"
        />
      </div>
    </div>
    <div
      className="flex-100 layout-row layout-wrap layout-align-center-start"
    >
      <div
        className="flex-100 layout-row layout-wrap layout-align-center-start file_header"
      >
        <p
          className="flex-100"
        >
          Uploaded
        </p>
      </div>
      <div
        className="flex-100 layout-row layout-wrap layout-align-center-start file_text"
      >
        <p
          className="flex-100"
        >
          19
        </p>
      </div>
    </div>
    <div
      className="flex-100 layout-row layout-wrap layout-align-center-start"
    >
      <div
        className="flex-100 layout-row layout-wrap layout-align-center-start file_header"
      >
        <p
          className="flex-100"
        >
          Status
        </p>
      </div>
      <div
        className="flex-100 layout-row layout-wrap layout-align-center-start file_text approved"
      >
        <p
          className="flex-100"
        >
          approved
        </p>
      </div>
    </div>
  </div>
  <div
    className="flex-100 layout-row layout-align-center-end"
  >
    <div
      className="upload_btn_wrapper flex-33 layout-row layout-align-center-center"
    >
      <form
        className="flex-none layout-row layout-align-center-center"
        onSubmit={[Function]}
      >
        <div
          className="flex-none"
          onClick={[Function]}
        >
          <i
            className="fa fa-pencil clip"
            style={
              Object {
                "background": "-webkit-linear-gradient(left, #333,#fafafa)",
              }
            }
          />
        </div>
        <input
          name="FOO_TYPE"
          onChange={[Function]}
          type="file"
        />
      </form>
    </div>
    <div
      className="upload_btn_wrapper flex-33 layout-row layout-align-center-center"
    >
      <div
        className="flex-none layout-row layout-align-center-center"
        onClick={[Function]}
      >
        <i
          className="clip fa fa-trash"
          style={
            Object {
              "background": "-webkit-linear-gradient(left, #333,#fafafa)",
            }
          }
        />
      </div>
    </div>
    <div
      className="upload_btn_wrapper flex-33 layout-row layout-align-center-center"
    />
  </div>
</div>
`;
